erpnext/regional/united_arab_emirates/utils.py

```diff
@@ -0,0 +1,14 @@
+from frappe.utils import flt
+from erpnext.controllers.taxes_and_totals import get_itemised_tax
+
+def update_itemwised_tax_data(doc):
+	if not doc.taxes: return
+
+	itemised_tax = get_itemised_tax(doc.taxes)
+
+	for row in doc.items:
+		tax_rate = sum([tax.get('tax_rate', 0) for d, tax in itemised_tax.get(row.item_code).items()])
+
+		row.tax_rate = flt(tax_rate, row.precision("tax_rate"))
+		row.tax_amount = flt((row.net_amount * tax_rate) / 100, row.precision("net_amount"))
```

Comment:
Don't calculate `tax_amount` again, it can lead to mismatched data due to precision. "itemised_tax" already have "tax_amount".